### ISSUE TITLE
Validate rack scale instance type input

### ISSUE THREAD

**Issue by nzoschke** [convox/rack#246]
I issued this command:

```
$ convox rack scale --count 6 --type m4xlarge
Name     convox
Status   updating
Version  20160109003156-cloudwatch
Count    6
Type     m4xlarge
```

And was surprised to see the stack update failed: "The specified instance type m4xlarge is not valid"

That's because it was a typo, the correct instance size is "m4.xlarge"

**Comment by awsmsrc** [comment]
@nzoschke @MiguelMoll can we create a list of valid choices here?

**Comment by mwarkentin** [comment]
There's a pretty good list here: http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html

**Comment by awsmsrc** [comment]
Are we wanting to allow all of these instance types?

**Comment by MiguelMoll** [comment]
Hard coding a list of types was an option we discussed a while back. Seems like the most straight forward approach. Would be nice to get a valid list programmatically.

Maybe a script (like the one used for ECS AMIs) could help here.

**Comment by mwarkentin** [comment]
It looks like boto has a list in here: https://github.com/boto/botocore/blob/develop/botocore/data/ec2/2016-04-01/service-2.json

```
"InstanceType":{
      "type":"string",
      "enum":[
        "t1.micro",
        "t2.nano",
        "t2.micro",
        "t2.small",
        "t2.medium",
        "t2.large",
        "m1.small",
        "m1.medium",
        "m1.large",
        "m1.xlarge",
        "m3.medium",
        "m3.large",
        "m3.xlarge",
        "m3.2xlarge",
        "m4.large",
        "m4.xlarge",
        "m4.2xlarge",
        "m4.4xlarge",
        "m4.10xlarge",
        "m2.xlarge",
        "m2.2xlarge",
        "m2.4xlarge",
        "cr1.8xlarge",
        "r3.large",
        "r3.xlarge",
        "r3.2xlarge",
        "r3.4xlarge",
        "r3.8xlarge",
        "x1.4xlarge",
        "x1.8xlarge",
        "x1.16xlarge",
        "x1.32xlarge",
        "i2.xlarge",
        "i2.2xlarge",
        "i2.4xlarge",
        "i2.8xlarge",
        "hi1.4xlarge",
        "hs1.8xlarge",
        "c1.medium",
        "c1.xlarge",
        "c3.large",
        "c3.xlarge",
        "c3.2xlarge",
        "c3.4xlarge",
        "c3.8xlarge",
        "c4.large",
        "c4.xlarge",
        "c4.2xlarge",
        "c4.4xlarge",
        "c4.8xlarge",
        "cc1.4xlarge",
        "cc2.8xlarge",
        "g2.2xlarge",
        "g2.8xlarge",
        "cg1.4xlarge",
        "d2.xlarge",
        "d2.2xlarge",
        "d2.4xlarge",
        "d2.8xlarge"
      ]
    },
```

**Comment by MiguelMoll** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;je5yHw5G&#x2F;483-validate-rack-scale-instance-type"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> Validate rack scale instance type</a>

**Comment by MiguelMoll** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;je5yHw5G&#x2F;483-validate-rack-scale-instance-type"><img src="https:&#x2F;&#x2F;github.trello.services&#x2F;images&#x2F;trello-icon.png" width="12" height="12"> Validate rack scale instance type</a>

**Comment by ddollar** [comment]
Fixed in #1040
